src/project-config.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as logging from 'plylog';
+
+const logger = logging.getLogger('cli.project-config');
+
+export interface ProjectConfigOptions {
+  root?: string;
+  entrypoint?: string;
+  shell?: string;
+  fragment?: string[];
+  fragments?: string[];
+}
+
+export class ProjectConfig {
```

Comment:
The difference is in the optionality and in fragment/fragments. One is produced by args parsing, the other passed to the API. I made a new one here optimized for calling from gulp... I'll remove this file for now and reconcile later. Sorry for including this.